js/src/edit-block/components/field-settings.js

```diff
@@ -50,8 +50,12 @@ const FieldSettings = ( {
 
 	return (
 		<>
-			{ control
+			{ control && control.settings
 				? control.settings.map( ( setting, index ) => {
+					if ( field.hasOwnProperty( 'parent' ) && 'location' === setting.name ) {
```

Comment:
Don't allow changing the location of a `sub_field`, as it's inside a field that has a location:

![the-sdf](https://user-images.githubusercontent.com/4063887/102303398-b306c200-3f20-11eb-84a1-6240e1fab45d.png)